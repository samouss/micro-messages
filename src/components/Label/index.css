.Label {
  border-radius: 5px;
  padding: 3px 7px;
  color: #FFF;
  font-weight: 600;
  font-size: 0.7rem;
  text-transform: uppercase;
}

.Label--Public {
  background-color: #225378;
}

.Label--Private {
  background-color: #DC3522;
}
