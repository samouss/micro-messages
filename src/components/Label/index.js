// @flow

import React from 'react';
import cx from 'classnames';
import './index.css';

export type Props = {
  type: 'public' | 'private',
};

const Label = ({ type }: Props) => {
  const isPublic = type === 'public';

  return (
    <span
      styleName={cx('Label', {
        'Label--Public': isPublic,
        'Label--Private': !isPublic,
      })}
    >
      {isPublic ? 'Public' : 'Prive'}
    </span>
  );
};

export default Label;
