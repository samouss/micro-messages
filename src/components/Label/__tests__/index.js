// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Label from '../index';
import type { Props } from '../index';

describe('<Label />', () => {
  it('expect to render the component correctly with "public"', () => {
    const props: Props = {
      type: 'public',
    };

    const component = shallow(
      <Label
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render the component correctly with "private"', () => {
    const props: Props = {
      type: 'private',
    };

    const component = shallow(
      <Label
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
