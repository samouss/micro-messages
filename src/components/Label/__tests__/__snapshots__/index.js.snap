// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> expect to render the component correctly with "private" 1`] = `
<span
  styleName="Label Label--Private"
>
  Prive
</span>
`;

exports[`<Label /> expect to render the component correctly with "public" 1`] = `
<span
  styleName="Label Label--Public"
>
  Public
</span>
`;
